dracut/image-boot/image-boot-generator

```diff
@@ -0,0 +1,83 @@
+#!/bin/sh
+# Copyright (C) 2016 Endless Mobile, Inc.
+# Licensed under the GPLv2
+#
+# Support booting from an image file hosted on a filesystem.
+# We parse the kernel parameters requesting this functionality and configure
+# other systemd units to go about setting up the block devices and mounting
+# the target filesystem.
+
+type getarg >/dev/null 2>&1 || . /lib/dracut-lib.sh
+
+GENERATOR_DIR="$1"
+
+image_device=$(getarg endless.image.device=)
+image_path=$(getarg endless.image.path=)
+
+[ -z "${image_device}" -o -z "${image_path}" ] && exit 0
+
+case "${image_device}" in
```

Comment:
true - we're not able to support making these sticks from XP unless we have the images totally baked ahead of time and we just DD them - I'd prefer to do that for ISOs rather than img.gz